src/ast.ls

```diff
@@ -2350,6 +2350,9 @@ class exports.For extends While
                 then "#idx #{ '<>'char-at pvar < 0 }#eq #tvar"
                 else "#pvar < 0 ? #idx >#eq #tvar : #idx <#eq #tvar"
         else
+            if @source instanceof Literal and @source.value is /^[0-9]+$/
```

Comment:
I thought so at first too, but all of the following are handled gracefully by JavaScript:
* 1e8
* 0.0
* true/false/null

This is the most general case I could find that triggers the syntax error. (Ah crap, I just found one more: negatives. I'll throw a `-?` in there.)